File: Java/src/main/java/org/codecop/dependencies/b/MarketingCampaign.java
package org.codecop.dependencies.b;

import java.time.DayOfWeek;
import java.time.LocalDateTime;

public class MarketingCampaign {

    public boolean isActive() {
        return milliSeconds() % 2 == 0;
    }

    /** only for testing */
    protected long milliSeconds() {
        return System.currentTimeMillis();
    }

    public boolean isCrazySalesDay() {
        return dayOfWeek().compareTo(DayOfWeek.FRIDAY) == 0;
    }

    /** only for testing */
    protected DayOfWeek dayOfWeek() {
        return LocalDateTime.now().getDayOfWeek();
    }
}

File: Kotlin/src/main/kotlin/org/codecop/dependencies/b/MarketingCampaign.kt
package org.codecop.dependencies.b

import java.time.DayOfWeek
import java.time.LocalDateTime

class MarketingCampaign {
    fun isActive(): Boolean {
        return milliSeconds() % 2 == 0L
    }

    private fun milliSeconds(): Long {
        return System.currentTimeMillis()
    }

    fun isCrazySalesDay(): Boolean {
        return dayOfWeek().compareTo(DayOfWeek.FRIDAY) == 0
    }

    protected fun dayOfWeek(): DayOfWeek {
        return LocalDateTime.now().dayOfWeek
    }
}